modules/tinymce/src/core/main/ts/keyboard/PageUpDownKeys.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Cell } from '@ephox/katamari';
+import Editor from '../api/Editor';
+import { NodeChangeEvent } from '../api/EventTypes';
+import { EditorEvent } from '../api/PublicApi';
+import VK from '../api/util/VK';
+import * as InlineBoundariesNavigation from './InlineBoundariesNavigation';
+import * as MatchKeys from './MatchKeys';
+
+const executeKeyupAction = (editor: Editor, evt: KeyboardEvent) => {
+  MatchKeys.execute([
+    { keyCode: VK.PAGE_UP, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, false) },
+    { keyCode: VK.PAGE_DOWN, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, true) }
+  ], evt);
+};
+
+const stopImmediatePropagation = (e: EditorEvent<NodeChangeEvent>) => e.stopImmediatePropagation();
+
+const isPageUpDown = (evt: EditorEvent<KeyboardEvent>) => {
+  return evt.keyCode === VK.PAGE_UP || evt.keyCode === VK.PAGE_DOWN;
+};
+
+const setNodeChangeBlocker = (blocked: Cell<boolean>, editor: Editor, block: boolean) => {
+  // Prevents registering multiple event blockers
+  if (block && !blocked.get()) {
+    editor.on('NodeChange', stopImmediatePropagation, true);
+  } else if (!block && blocked.get()) {
+    editor.off('NodeChange', stopImmediatePropagation);
+  }
+
+  blocked.set(block);
+};
+
+const setup = (editor: Editor) => {
```

Comment:
I mentioned this in a previous comment, but can you please add a comment here, stating why we are using `keyup` instead of `keydown` so we know in future why this decision was made (as it's inconsistent with the other overrides).

Reply:
This has not been done, all you did was move the previous comment which only explains the `NodeChange` prevention. It has nothing about keydown/keyup.

Reply:
This description is not quite correct sorry. See Spocke's previous explanation for why we aren't doing this when I brought it up. eg determining the correct placement on keydown is very complicated and would require handling many edge cases, which we don't have the resources to handle currently. As such we allow the browser to change the selection and then make adjustments later.

You've also now lost the reason we had the `NodeChange` override as well.